### PR TITLE
[jit][ocl] Debug and build kernel improved

### PR DESCRIPTION
**Pull request by jjfumero**
#### Description

Build log for OpenCL fixed. 

#### Problem description

There was an issue when the OpenCL build was not correct, the log string was not printed. 

#### Backend/s tested

Mark the backends affected by this PR.

- [X] OpenCL
- [ ] PTX
- [ ] SPIRV

#### OS tested

Mark the OS where this PR is tested.

- [X] Linux
- [ ] OSx
- [ ] Windows

#### Did you check on FPGAs?

If it is applicable, check your changes on FPGAs.

- [ ] Yes
- [X] No

#### How to test the new patch?

```bash
make 
make tests
```

### CODE REVIEW

File: tornado-drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/enums/OCLBuildStatus.java [hunk 1]
```diff
@@ -24,35 +24,30 @@
...
 public enum OCLBuildStatus {
-    CL_BUILD_SUCCESS(0), CL_BUILD_NONE(-1), CL_BUILD_ERROR(-2), CL_BUILD_IN_PROGRESS(-3), CL_BUILD_UNKNOWN(-4);
+    CL_BUILD_SUCCESS(0), //
+    CL_BUILD_NONE(-1), //
+    CL_BUILD_ERROR(-2), //
+    CL_BUILD_IN_PROGRESS(-3), //
+    CL_BUILD_UNKNOWN(-4);
 
-    private final int value;
+    private final int errorCode;
```
Comment (mairooni) [inline tornado-drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/enums/OCLBuildStatus.java:33]: Is the name `errorCode` appropriate? The build might be successful. How about something like `buildCode`?

Comment (stratika) [reply tornado-drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/enums/OCLBuildStatus.java:33]: Yes, we can rename the variable name to `buildCode` to include also the successful build code.

**Comment by jjfumero** [comment]
You can try to compile something unsupported in Java.

File: tornado-drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/OCLCodeCache.java [hunk 2]
```diff
@@ -558,6 +557,57 @@ private boolean isInputSourceSPIRVBinary(byte[] source) {
...
+        file = new File(STR."\{outDir}/\{identifier}\{OPENCL_SOURCE_SUFFIX}");
+        try (FileOutputStream fos = new FileOutputStream(file)) {
+            fos.write(source);
+        } catch (IOException e) {
+            error("unable to write error log: ", e.getMessage());
+        }
+
+    }
+
+    private void dumpOpenCLKernel(String id, String entryPoint, String log, byte[] source) {
```
Comment (stratika) [inline tornado-drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/OCLCodeCache.java:580]: Is this method used? If yes, what is the difference between this and the other `dumpKernelSource`?

Comment (jjfumero) [reply tornado-drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/OCLCodeCache.java:580]: sorry, that was a leftover from a refactoring I did. I will remove it

**Comment by jjfumero** [comment]
All comments addressed.

**Review by stratika — APPROVED** [review]
LGTM

**Review by mairooni — APPROVED** [review]

**Pull request merged**